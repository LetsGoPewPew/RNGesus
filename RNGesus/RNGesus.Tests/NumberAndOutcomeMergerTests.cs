using Model;
using NUnit.Framework;
using BusinessLogic;
using ApprovalTests;
using ApprovalTests.Reporters;
using System.Text;
using System.Collections.Generic;
using ApprovalTests.Reporters.Windows;

namespace RNGesus.Tests
{
    [UseReporter(typeof(VisualStudioReporter))]
    [TestFixture]
    class NumberAndOutcomeMergerTests
    {
        private List<WeightedNamedOutcome> outcomes = new List<WeightedNamedOutcome>();
        private NumberAndOutcomeMerger numberAndOutcomeMerger;

        [OneTimeSetUp]
        public void init()
        {
            outcomes.Add(new WeightedNamedOutcome("Stian1", 1, true));
            outcomes.Add(new WeightedNamedOutcome("Stian2", 2, true));
            outcomes.Add(new WeightedNamedOutcome("Stian3", 3, true));
            outcomes.Add(new WeightedNamedOutcome("Stian4", 4, true));
            outcomes.Add(new WeightedNamedOutcome("Stian5", 5, true));
            outcomes.Add(new WeightedNamedOutcome("Stian6", 6, true));
            outcomes.Add(new WeightedNamedOutcome("Stian7", 7, true));
            outcomes.Add(new WeightedNamedOutcome("Stian8", 8, true));
            outcomes.Add(new WeightedNamedOutcome("Stian9", 9, true));
            outcomes.Add(new WeightedNamedOutcome("Stian10", 10, true));

            numberAndOutcomeMerger = new NumberAndOutcomeMerger(new List<int>(), outcomes);
        }
        
        [Test]
        public void Assert_that_range_values_are_correct()
        {
            StringBuilder result = new StringBuilder();
            
            foreach(WeightedNamedOutcome outcome in numberAndOutcomeMerger.GetOutcomes())
            {
                result.AppendLine(outcome.GetRange().ToString() + "  ---  " + outcome.Name);
            }

            Approvals.Verify(result);
        }
    }
}
