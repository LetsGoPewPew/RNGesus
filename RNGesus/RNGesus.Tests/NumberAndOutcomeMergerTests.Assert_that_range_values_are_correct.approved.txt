From: 1 To: 2  ---  Stian1
From: 2 To: 4  ---  Stian2
From: 4 To: 7  ---  Stian3
From: 7 To: 11  ---  Stian4
From: 11 To: 16  ---  Stian5
From: 16 To: 22  ---  Stian6
From: 22 To: 29  ---  Stian7
From: 29 To: 37  ---  Stian8
From: 37 To: 46  ---  Stian9
From: 46 To: 56  ---  Stian10
